sys/drv/pit.c

```diff
@@ -100,11 +118,14 @@ static int timer_pit_stop(timer_t *tm) {
 static bintime_t timer_pit_gettime(timer_t *tm) {
   device_t *dev = device_of(tm);
   pit_state_t *pit = dev->state;
-  uint64_t count = pit_get_counter64(pit);
   uint32_t freq = pit->timer.tm_frequency;
-  uint32_t sec = count / freq;
-  uint32_t frac = count % freq;
-  bintime_t bt = bintime_mul(HZ2BT(freq), frac);
+  uint32_t sec, ticks;
+  WITH_INTR_DISABLED {
+    pit_update_time(pit);
```

Comment:
Why don't you handle overflow here as you do in `pit_intr`? It can't occur here or is there some other explanation?

Reply:
Cause in pit_intr we are sure that the count_period already past (we just unsure how many) and we want to check if it is already added to ticks.